docs/guide.rst

```diff
@@ -31,109 +31,134 @@ The `WGPU <https://github.com/pygfx/wgpu-py>`_ library provides the low level AP
+Pygfx is structured like many other rendering engines. We create a `scene` (a
+worlds/scenarios to render) using three main ingredients: (1) `objects` and
+their visual properties, (2) `light` sources, and (3) a `camera` that sees
+things. Once we defined those three things, we can position them in our scene
+and then use a `renderer` to look at what we have created (or take
+pictures/videos of it).
 
-.. code-block:: python
+.. note:: 
+    The :ref:`full code example <full_example>` can be found below.
 
-    scene = gfx.Scene()
+Let's look at a hello world example of how this works: A rotating cube. We begin
+by defining an empty `Scene`::
 
-    geometry = gfx.BoxGeometry(200, 200, 200)
-    material = gfx.MeshPhongMaterial(color=(1, 1, 0, 01))
-    cube = gfx.Mesh(geometry, material)
+    import pygfx as gfx
 
-    scene.add(cube)
+    scene = gfx.Scene()
 
+Right now this scene is very desolate. There is no light, no objects, and
+nothing that can look at those objects. Let's change this by adding some
+`light` and a `camera`::
 
-Opening a window to render to
------------------------------
+    # and god said ...
+    scene.add(gfx.AmbientLight())
 
-Your visualization must end up on the screen somehow. For this, we use the
-canvas abstraction provided by wgpu-py.
+    camera = gfx.PerspectiveCamera(70, 16 / 9)
+    camera.position.z = 400
 
-.. code-block:: python
+Now there is light and a camera to perceive the light. To complete the setup
+we also need to add an object to look at::
 
-    # Create Qt widget that can function as a canvas
-    from wgpu.gui.qt import WgpuCanvas
-    canvas = WgpuCanvas()
+    geometry = gfx.box_geometry(200, 200, 200)
+    material = gfx.MeshPhongMaterial(color=(1, 1, 0, 1))
+    cube = gfx.Mesh(geometry, material)
 
+    scene.add(cube)
 
-We can ask the canvas to schedule a draw event, and tell it what to do
-to perform a draw.
+Objects are slightly more complicated than lights or cameras. They have a
+`geometry`, which controlls an object's form, and a `material`, which controls
+an object's appearance (color, reflectiveness, etc).
 
-.. code-block:: python
+Now we have all the necessary ingredients and it is time to take a look. This is
+done by a `renderer` and happens in a two step process known as a draw call:
+First, the renderer looks through the camera we created earlier, paints an image
+of what it sees, and stores it in a intermediate buffer. Second, the renderer
+takes this buffer and hands it over to a `canvas` which is responsible for
+displaying the image (here inside an on-screen window)::
 
-    def animate():
-       ...  # we'll get to this
+    from wgpu.gui.auto import WgpuCanvas, run
 
+    canvas = WgpuCanvas()
+    renderer = gfx.renderers.WgpuRenderer(canvas)
 
-    canvas.request_draw(animate)
+    def draw_function():
+        # update the internal buffer
+        renderer.render(scene, camera)
 
+        # Note: the handover to the canvas is implicit
 
```

Comment:
The internal buffer is already a responsibility of the canvas. Also at the low level (the canvas provides the texture to render to). So the handover is already inside `renderer.render()`.
```suggestion
```